### PR TITLE
[ticket/13414] Set Content-Length header only if status is not 304

### PR DESCRIPTION
**Pull request by marc1706**
Ticket: https://tracker.phpbb.com/browse/PHPBB3-13414

### CODE REVIEW

**Comment by bantu** [comment]
You're sending a header after sending the complete body.

**Comment by marc1706** [comment]
Oh, should be before flush(). Then again, I guess the previous location was also fine (right after the beginning of the if)?

**Comment by bantu** [comment]
@marc1706 Yes, that's where it has to go for olympus. ascraeus was a different story.

**Comment by marc1706** [comment]
@bantu updated PR and also created an additional one for ascraeus

File: phpBB/download/file.php [hunk 1]
```diff
@@ -509,16 +509,17 @@ function send_file_to_browser($attachment, $upload_dir, $category)
...
-	{
-		header("Content-Length: $size");
-	}
-
 	// Close the db connection before sending the file
 	$db->sql_close();
 
 	if (!set_modified_headers($attachment['filetime'], $user->browser))
 	{
+		// Send Content-Length after delivering file
```
Comment (bantu) [inline phpBB/download/file.php]: "after delivering file" ?

**Comment by marc1706** [comment]
@bantu Should be better now :)

**Pull request merged**